core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala

```diff
@@ -186,9 +187,13 @@ trait DeltaConfigsBase extends DeltaLogging {
 
     val globalConfs = entries.asScala.flatMap { case (key, config) =>
       val sqlConfKey = sqlConfPrefix + config.key.stripPrefix("delta.")
-      Option(sqlConfs.getConfString(sqlConfKey, null)) match {
-        case Some(default) => Some(config(default))
-        case _ => None
+      val sqlConf = Option(sqlConfs.getConfString(sqlConfKey, null))
+      val deprecatedSqlConfKey = deprecatedSqlConfPrefix + config.key.stripPrefix("delta.")
+      val deprecatedSqlConf = Option(sqlConfs.getConfString(deprecatedSqlConfKey, null))
+      (sqlConf, deprecatedSqlConf) match {
+        case (Some(default), _) => Some(config(default))
```

Comment:
You need to handle the case where they are BOTH set.

If they are both set ....
- and both have SAME value -> okay
- and both have DIFFERENT value -> need to throw a nice error explaining that this is ambiguous and is not allowed